syntax = "proto3";

package ocr;

message ImageTask {
  int32 id = 1;
  bytes image_data = 2;
}

message BatchRequest {
  repeated ImageTask tasks = 1;
}

message BatchResult {
  int32 id = 1;
  string text = 2;
  int64 processing_time_ms = 3;
}

message BatchResponse {
  repeated BatchResult results = 1;
}

service OCRService {
  rpc ProcessBatch (BatchRequest) returns (BatchResponse);
}
